06_puppy/pfannerj/storer_test.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+const firstPuppyID uint32 = 1
+
+var (
+	firstPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Retriever",
+			Colour: "Golden",
+			Value:  9999.99,
+		}
+	}
+	modifiedPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Bitsa",
+			Colour: "Mixed",
+			Value:  1.99,
+		}
+	}
+	anotherPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Labrador",
+			Colour: "Black",
+			Value:  999.99,
+		}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store     Storer
+	thisStore func() Storer
+}
+
+func (s *storerSuite) SetupTest() {
+	// create test store and add the first puppy
+	s.store = s.thisStore()
+	puppy := firstPuppy()
+	_, err := s.store.CreatePuppy(puppy)
+	if err != nil {
+		panic("Failed to setup puppy test")
+	}
+}
+
+func TestStorer(t *testing.T) {
+
+	suite.Run(t, &storerSuite{
+		thisStore: func() Storer { return &SyncStore{} },
```

Comment:
i'd call it newStore, but that's a bit nitpicky ;)

Reply:
Likewise will rename it in my Lab7.... but, I have a question on this... you had suggested previously that this line could be written like this...
`suite.Run(t, &storerSuite{thisStore: thisStore.NewSyncStore()})`
I couldn't get the above to work?? yet it looks equivalent to me...